openapi-cli/src/main/java/org/ballerinalang/ballerina/service/OpenApiResourceMapper.java

```diff
@@ -372,30 +366,22 @@ private void handleRecordPayload(RequiredParameterNode queryParam, Map<String, M
         String componentName = typeSymbol.name().trim();
         ModelImpl messageModel = new ModelImpl();
         if (typeSymbol instanceof TypeReferenceTypeSymbol) {
-            TypeReferenceTypeSymbol typeRef =
-                    (TypeReferenceTypeSymbol) typeSymbol;
+            TypeReferenceTypeSymbol typeRef = (TypeReferenceTypeSymbol) typeSymbol;
             // Handle record type request body
             if (typeRef.typeDescriptor() instanceof RecordTypeSymbol) {
-                RecordTypeSymbol recordTypeSymbol =
-                        (RecordTypeSymbol) typeRef.typeDescriptor();
-                List<FieldSymbol> rfields =
-                        recordTypeSymbol.fieldDescriptors();
+                RecordTypeSymbol recordTypeSymbol = (RecordTypeSymbol) typeRef.typeDescriptor();
+                List<FieldSymbol> rfields = recordTypeSymbol.fieldDescriptors();
                 for (FieldSymbol field: rfields) {
                     String type = field.typeDescriptor().typeKind().toString().toLowerCase(Locale.ENGLISH);
                     Property property = getOpenApiProperty(type);
                     if (type.equals(Constants.TYPE_REFERENCE) && property instanceof RefProperty) {
                         RefModel refModel = new RefModel();
                         refModel.setReference(field.typeDescriptor().name().trim());
                         ((RefProperty) property).set$ref(refModel.get$ref());
-
-                        Optional<Symbol> recordSymbol = semanticModel.symbol(queryParam.syntaxTree().filePath(),
-                                        LinePosition.from(field.location().lineRange().startLine().line(),
-                                                field.location().lineRange().startLine().offset()));
-
-                        VariableSymbol recordVariable = (VariableSymbol) recordSymbol.orElseThrow();
-                        if (recordVariable.typeDescriptor() instanceof TypeReferenceTypeSymbol) {
-                            TypeReferenceTypeSymbol typeRecord =
-                                    (TypeReferenceTypeSymbol) recordVariable.typeDescriptor();
+                        Optional<TypeSymbol> recordSymbol = semanticModel.type(field.location().lineRange());
+                        TypeSymbol recordVariable =  recordSymbol.orElseThrow();
+                        if (recordVariable instanceof TypeReferenceTypeSymbol) {
```

Comment:
Consider using `type.kind()` for the validations, instead of using `instanceof` checks.